src/tree/SubscriptionTreeItem.ts

```diff
@@ -40,7 +40,17 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         return await this.createTreeItemsWithErrorHandling(
             staticWebApps,
             'invalidStaticWebApp',
-            ss => new StaticWebAppTreeItem(this, ss),
+            ss => {
+                try {
+                    return new StaticWebAppTreeItem(this, ss);
+                } catch (error) {
+                    const parsedError: IParsedError = parseError(error);
+                    if (/(null|undefined).*repositoryUrl/.test(parsedError.message)) {
```

Comment:
We're the ones throwing this error, so you don't need to catch and parse it. Just change the `nonNullProp` [here](https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/tree/StaticWebAppTreeItem.ts#L42) to something like
```
if (this.data.repositoryUrl) {
    this.repositoryUrl = this.data.repositoryUrl;
} else {
    throw new Error(onlyGitHubSupported);
}
```